### PR TITLE
FAT-1037 Hold request successfully created when request policy and item status allow it

### PR DESCRIPTION
**Pull request by bisecomsergiy**
### Purpose
Increase mod-circulation coverage by integration tests, decrease tech debt

### Approach
Created karate test for Requests: Given an item Id, a user Id, and a pickup location, attempt to create a hold request when the applicable request policy allows holds and item status is not "Available", "Recently returned", "In process (not requestable)", "Declared lost", "Lost and paid", "Aged to lost", "Claimed returned", "Missing from ASR", "Long missing", "Retrieving from ASR", "Withdrawn", "Order closed", "Intellectual item", "Unavailable", or "Unknown"

### Learning
https://issues.folio.org/browse/FAT-1037

### CODE REVIEW

**Review by viktor-draban — APPROVED** [review]

File: mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature [hunk 1]
```diff
@@ -464,3 +464,51 @@ Feature: Requests tests
     * def extRequestId = call uuid1
     * def extRequestType = 'Recall'
     * call read('classpath:vega/mod-circulation/features/util/initData.feature@PostRequest') { requestId: #(extRequestId), itemId: #(extItemId), requesterId: #(extUserId), extRequestType: #(extRequestType), extInstanceId: #(instanceId), extHoldingsRecordId: #(holdingId) }
+
+    # This scenario does not cover testing for item with status 'Available in ASR' due to lack of implementation
+  Scenario Outline: Requests: Given an item Id, a user Id, and a pickup location, attempt to create a hold request when the applicable request policy allows holds and item status is not "Available", "Recently returned", "In process (not requestable)", "Declared lost", "Lost and paid", "Aged to lost", "Claimed returned", "Missing from ASR", "Long missing", "Retrieving from ASR", "Withdrawn", "Order closed", "Intellectual item", "Unavailable", or "Unknown"
```
Comment (yusuf-murodov-epam) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature:469]: This test is not to check failing cases, please write passing cases for all not provided statuses as FAT-1038

Comment (bisecomsergiy) [reply mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature:469]: Sorry, did not understand you. Could you indicate line number of the test asserts failing cases?

Comment (yusuf-murodov-epam) [reply mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature:469]: please check this link [here](https://folio-project.slack.com/archives/CD1CCKE4V/p1643211701164700)

Comment (bisecomsergiy) [reply mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature:469]: I have checked the correspondence. And the test  is about passing cases for all not provided statuses..

File: mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature [hunk 2]
```diff
@@ -464,3 +464,51 @@ Feature: Requests tests
...
+    * requestEntityRequest.requesterId = extUserId
+    * requestEntityRequest.requestType = 'Hold'
+    * requestEntityRequest.holdingsRecordId = holdingId
+    * requestEntityRequest.requestLevel = 'Item'
+    Given path 'circulation', 'requests'
+    And request requestEntityRequest
+    When method POST
+    Then status 201
+    And match response.item.barcode == <itemBarcode>
+    And match response.requester.barcode == <userBarcode>
```
Comment (yusuf-murodov-epam) [inline mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature:495]: use the @PostRequest in initData.future

Comment (bisecomsergiy) [reply mod-circulation/src/main/resources/vega/mod-circulation/features/requests.feature]: It is done, thank you

**Pull request merged**